apps/meteor/ee/server/lib/audit/endpoints.ts

```diff
@@ -0,0 +1,95 @@
+import type { IUser, IRoom } from '@rocket.chat/core-typings';
+import { Rooms, AuditLog } from '@rocket.chat/models';
+import type { PaginatedRequest, PaginatedResult } from '@rocket.chat/rest-typings';
+import Ajv from 'ajv';
+
+import { API } from '../../../../app/api/server/api';
+import { getPaginationItems } from '../../../../app/api/server/helpers/getPaginationItems';
+import { findUsersOfRoom } from '../../../../server/lib/findUsersOfRoom';
+
+const ajv = new Ajv({
+	coerceTypes: true,
+});
+
+type AuditRoomMembersParams = PaginatedRequest<{
+	roomId: string;
+	filter: string;
+}>;
+
+const auditRoomMembersSchema = {
+	type: 'object',
+	properties: {
+		roomId: { type: 'string', minLength: 1 },
+		filter: { type: 'string' },
+		count: { type: 'number' },
+		offset: { type: 'number' },
+		sort: { type: 'string' },
+	},
+	required: ['roomId'],
+	additionalProperties: false,
+};
+
+export const isAuditRoomMembersProps = ajv.compile<AuditRoomMembersParams>(auditRoomMembersSchema);
```

Comment:
This is quite confusing for me as well. We have the exact same case as discussed above (about the place where this file should live), and again, since we don't have that "pattern" defined, we do have a bunch of ajv validation for EE endpoints living in the rest-typings folder, I'd go with the "pattern" we are using until now.

Reply:
I'd prefer to bring @tassoevan around since the last time i did the opposite I was told to do this way and would prefer to have an standard def rather than 2 diff opinions.

Reply:
From Tasso's old discussion:

```
It's something you're trying to push more. The augmentation itself is needed because some APIs were born to be extensible (e.g. API.addRoute, MessageAction.add etc.)
Having a single place to drop all the signatures is impractical as we have been splitting the source code in packages and sometimes behind a flag (e.g. Enterprise code)
Our first attempt to enforce typings on APIs like it was with the REST API endpoints
A single interface held all the call signatures
Then Pierre started working on moving the models to new packages
It imposed a challenged the we haven't thinking at that point: how to split the interfaces and the implementations?
That's why we have models and model-typings
As we learned that type imports are a thing, it was obvious it could be only models
Therefore a module that can't import any model still can do import type { ... } from '@rocket.chat/models'
Without changing the final JS output
So we thought "what would happen if we create a package adding more endpoints to the API?"
:hmmm:
1
Then we merged two concepts: (1) augment interfaces while (2) importing implementations

Reply:
@MarcosSpessatto this is kinda related to the last thread on the architecture channel (from @pierre-lehnen-rc). For now EE endpoints don't have a definitive place to be defined (types and with schema validations). There is an intent of making a single package for FOSS endpoints, Meteor methods, and streams, letting non-FOSS splitted elsewhere by domain (e.g. a `@rocket.chat/rest-typings-ee` should **NOT** exist, but those types and schemas could reside in something like `@rocket.chat/auditing`).

Reply:
Awesome, thanks for the explanation guys!